src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -380,12 +401,35 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 	{
 #if PG_VERSION_NUM >= PG_VERSION_16
 		DefElem *opt = NULL;
+		 int opt_count = 0;
 		foreach_ptr(opt, stmt->opt)
 		{
-			if (strcmp(opt->defname, "admin") == 0)
-			{
-				appendStringInfo(buf, " WITH ADMIN OPTION");
+			switch (opt->defname)
+        {
+            case "admin":
+                appendStringInfo(buf, " WITH ADMIN OPTION");
+                opt_count++;
+                break;
+
+			case "inherit":
+				if (opt_count > 0)
+				{
+					appendStringInfo(buf, ", ");
+				}
+				appendStringInfo(buf, "INHERIT OPTION ");
+				opt_count++;
+				break;
+
+				
+        	case "set":
+  				if (opt_count > 0)
+				{
+					appendStringInfo(buf, ", ");
+				}
+				appendStringInfo(buf, "SET OPTION ");
```

Comment:
Same as for inherit, you need to check the value of "set" option - it could be false. Note that TRUE is the default.